### PR TITLE
Add a blog post about "Remote Builds without the Bytes"

### PR DESCRIPTION
**Pull request by buchgr**
Add a blog post about "Remote Builds without the Bytes" in Bazel 0.25.

@EricBurnett @agoulti @ishikhman

### CODE REVIEW

File: _posts/2019-05-07-builds-without-bytes.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+ 
+When using Bazel with remote caching and execution it downloads every output file of every action executed (or cached) to the host machine. The total size of all output files in a build will often be in the tens and even hundreds of gigabytes and make the network bandwidth the limiting factor. We know from experience that most of the time developers are only interested in a small subset of the output files produced during a build i.e. the final binary and quite often in a continuous integration build the output files are of no interest at all.
+ 
+We have thus been working on virtualizing action outputs in Bazel so to only transfer file metadata, that is content hash and size, instead of file contents. This will significantly reduce the amount of data being transferred between Bazel and the remote caching or execution system. Our own testing and early user reports suggest that double digest build speed improvements are the common case. The specific gains for your project will mostly depend on the available network bandwidth, latency and size of build outputs.
```
Comment (aiuto) [inline _posts/2019-05-07-builds-without-bytes.md]: I can't parse this:  "that double digest build speed improvements "

Comment (buchgr) [reply _posts/2019-05-07-builds-without-bytes.md]: Ahh a typo. I rephrased it. Done.

File: _posts/2019-05-07-builds-without-bytes.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+$ bazel build :foo --experimental_remote_download_outputs=minimal
+$ bazel shutdown
+# This will be clean build, because the metadata from the previous build are
+# not yet properly persisted to disk
+$ bazel build :foo --experimental_remote_download_outputs=minimal
+```
+ 
+Incremental builds that don't involve a shutdown will work fine. Please follow [#8248](https://github.com/bazelbuild/bazel/issues/8248) for the current state, progress updates and to express interest.
+ 
+### Local file uploads and URI rewriting in the Build Event Service is disabled
```
Comment (agoulti) [inline _posts/2019-05-07-builds-without-bytes.md]: What is the end user experience of this?

Comment (buchgr) [reply _posts/2019-05-07-builds-without-bytes.md]: Files will not be available in RS

**Comment by EricBurnett** [comment]
Apologies for the delay reviewing - couple quick comments. Though in the future, a slower turn-around would be appreciated - 24h from authoring to posting publicly doesn't leave too much time for getting the relevant eyes on it and approvals and whatnot. Thanks in advance!

 - Overall, well written. Nicely done.
 - How convinced are you that this is usable in its current form? For most RBE customers it seems that at least one of (toplevel), (BES support) is needed for CI build results to be useful, and so I don't have any examples of customers having turned it on for more than quick validation yet. I wonder if we want to surface these limitations higher, e.g. present this as as _early access_ to a feature aiming for broad usability by 0.26, seeking feedback and real-world use-cases? (Assuming these are both landing in 0.26; otherwise later.)
 - Similarly, should we call out higher that this is _experimental_ support, from a level-of-testing perspective? It's implied by the flag names, but the framing of the blog post makes it sound like a full, ready-to-use feature. I'm thinking about things like hybrid remote/local builds, coverage of all the different types of builtin rules, etc. (Not a concern if you've robustly tested these scenarios already, I just know that bazel has a whole lot of dark corners that aren't easily exercised).

**Comment by buchgr** [comment]
> How convinced are you that this is usable in its current form?

Very :). I have sent you an e-mail about the BES part.

> Similarly, should we call out higher that this is experimental support, from a level-of-testing perspective? It's implied by the flag names, but the framing of the blog post makes it sound like a full, ready-to-use feature.

Sure. I ll update the post.

> I'm thinking about things like hybrid remote/local builds, coverage of all the different types of builtin rules, etc. (Not a concern if you've robustly tested these scenarios already, I just know that bazel has a whole lot of dark corners that aren't easily exercised).

I am not concerned about this. I think this has been adequately tested. Hybrid local remote execution should work fine and I have also turned it on our own Buildkite CI and it seems to work well.

**Pull request merged**